plugins/status/metrics.go

```diff
@@ -1,82 +1,168 @@
 			Name: "last_success_bundle_activation",
-			Help: "Gauge for the last success bundle activation."},
+			Help: "Gauge for the last success bundle activation.",
+		},
 		[]string{"name", "active_revision"},
 	)
-	lastSuccessfulDownload = prometheus.NewGaugeVec(
+	lastSuccessfulDownload := prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
 			Name: "last_success_bundle_download",
-			Help: "Gauge for the last success bundle download."},
+			Help: "Gauge for the last success bundle download.",
+		},
 		[]string{"name"},
 	)
-	lastSuccessfulRequest = prometheus.NewGaugeVec(
+	lastSuccessfulRequest := prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
 			Name: "last_success_bundle_request",
-			Help: "Gauge for the last success bundle request."},
+			Help: "Gauge for the last success bundle request.",
+		},
 		[]string{"name"},
 	)
-	bundleLoadDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+
+	bundleLoadDuration := newBundleLoadDurationCollector(prometheusConfig)
+
+	return &collectors{
+		opaInfo:                  opaInfo,
+		pluginStatus:             pluginStatus,
+		loaded:                   loaded,
+		failLoad:                 failLoad,
+		lastRequest:              lastRequest,
+		lastSuccessfulActivation: lastSuccessfulActivation,
+		lastSuccessfulDownload:   lastSuccessfulDownload,
+		lastSuccessfulRequest:    lastSuccessfulRequest,
+		bundleLoadDuration:       bundleLoadDuration,
+	}
+}
+
+func newBundleLoadDurationCollector(prometheusConfig *PrometheusConfig) *prometheus.HistogramVec {
+	return prometheus.NewHistogramVec(prometheus.HistogramOpts{
 		Name:    "bundle_loading_duration_ns",
 		Help:    "Histogram for the bundle loading duration by stage.",
-		Buckets: prometheus.ExponentialBuckets(1000, 2, 20),
+		Buckets: prometheusConfig.BundleLoadDurationNanoseconds.Buckets,
 	}, []string{"name", "stage"})
+}
 
-	// allCollectors is a list of all collectors maintained by the status plugin.
-	// Note: when adding a new collector, make sure to also add it to this list,
-	// or it won't survive status plugin reconfigure events.
-	allCollectors = []prometheus.Collector{
-		opaInfo,
-		pluginStatus,
-		loaded,
-		failLoad,
-		lastRequest,
-		lastSuccessfulActivation,
-		lastSuccessfulDownload,
-		lastSuccessfulRequest,
-		bundleLoadDuration,
+func (c *collectors) RegisterAll(register prometheus.Registerer, logger logging.Logger) {
+	if register == nil {
+		return
 	}
-)
+	for _, collector := range c.toList() {
+		if err := register.Register(collector); err != nil {
+			logger.Error("Status metric failed to register on prometheus :%v.", err)
+		}
+	}
+}
 
-func init() {
-	opaInfo.Set(1)
+func (c *collectors) UnregisterAll(register prometheus.Registerer) {
+	if register == nil {
+		return
+	}
+
+	for _, collector := range c.toList() {
+		register.Unregister(collector)
+	}
+}
+
+func (c *collectors) ReregisterCustomizedCollectors(register prometheus.Registerer, config *PrometheusConfig, logger logging.Logger) {
```

Comment:
Currently we only update the bundleLoadDuration collector so we can change the function name to reflect that.